### DISCUSSION TITLE
Dynamic ENV variables

### DISCUSSION THREAD

**Discussion by anstapol** [vercel/community#4752] [Help] [answered]
### Summary

Hi guys,

We have a Next.js app on Vercel which talks to the backend API on different platform. Most of the times when we want to test frontend updates, the web-app makes requests to a staging API server except when there's been some changes on how that API responds, that's when we need to add a new environment variable specific API url which supports that feature.

I was wondering if there was any way to maybe inject that environment variable on build phase? 
Ideally we'd build custom API staging URL using actual branch name, make a call to a health check route and in case it fails just use the default staging url.

Would appreciate any ideas :) 

### Steps to Reproduce

It's not really a problem so no steps to reproduce 😅

**Comment by amyegan** [comment, marked as answer]
There are a few ways to set and change environment variables and related config options. I'm not sure which of these would fit best in your workflow, so I'll just share the documentation links below. 😄 

- [Environment Variables](https://vercel.com/docs/projects/environment-variables#) can be managed in Project Settings for automatic deployments, but you would need to know the URL before starting the build.
- Environment variables can also be set per deployment if you use the CLI to deploy. You can do this by using the [`vercel deploy`](https://vercel.com/docs/cli/deploy#) command with the [`--build-env`](https://vercel.com/docs/cli/deploy#build-env) or [`--env`](https://vercel.com/docs/cli/deploy#env) option. This can work well with [CI workflows](https://vercel.com/guides/using-vercel-cli-for-custom-workflows).
- [Vercel Edge Config](https://vercel.com/docs/storage/edge-config#) can be used in Edge Functions, Edge Middleware, and Serverless Functions.
